src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import './GoodsList.scss';
+import '../../styles/button.scss';
+
+interface Props {
+  goods: string[],
+}
+
+interface State {
+  goods: string[],
+  sortBy: string,
+}
+
+export class GoodsList extends React.Component<Props> {
+  state = {
+    goods: [...this.props.goods],
+    sortBy: 'alphab',
+  };
+
+  reverse = () => (
+    this.setState((state: State) => ({
+      goods: state.goods.reverse(),
+    }))
+  );
+
+  sortAlphab = () => {
+    this.setState({
+      sortBy: 'alphab',
+    });
+
+    this.sort();
+  };
+
+  length = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+
+    this.sort();
+  };
+
+  reset = () => {
+    this.setState({
+      goods: [...this.props.goods],
+    });
+  };
+
+  sort = () => (
+    this.setState((state: State) => ({
+      goods: state.goods.sort(
+        (productA, productB) => {
+          switch (state.sortBy) {
+            case 'alphab': {
+              return productA.localeCompare(productB);
+            }
+
+            case 'length': {
+              return productA.length - productB.length;
+            }
+
+            default: {
+              return 0;
+            }
+          }
+        },
+      ),
+    }))
+  );
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="goods-list">
+        <h1 className="goods-list__title">
+          Goods list
+        </h1>
+
+        <div className="goods-list__buttons">
+          <button
+            type="button"
+            className="goods-list__button button"
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          <button
+            type="button"
+            className="goods-list__button button"
+            onClick={this.sortAlphab}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className="goods-list__button button"
+            onClick={this.reset}
+          >
+            Reset
+          </button>
+
+          <button
+            type="button"
+            className="goods-list__button button"
+            onClick={this.length}
+          >
+            Sort by length
+          </button>
+        </div>
+
+        <ul className="goods-list__list">
+          {goods.map((product, index) => (
+            <li
+              className="goods-list__list-item"
+              key={`${product} ${index * Math.random()}`}
```

Comment:
you can just use the product as a key

Reply:
I'm advisedly used Math.random  here, because, what if in this array will arise a two equals products?

Reply:
please, kindly, answer me am I right or not?)